test/unittest/unittest_nntrainer_tensor.cpp

```diff
@@ -448,6 +448,204 @@ TEST(nntrainer_Tensor, add_i_01_n) {
+    EXPECT_EQ(t, answer);
+  }
+  {
+    nntrainer::Tensor t = ranged(3, 2, 4, 5);
+    nntrainer::Tensor m = ranged(1, 1, 1, 5);
+    float answer_data[] = {
+      0,   2,   4,   6,   8,   5,   7,   9,   11,  13,  10,  12,  14,  16,
+      18,  15,  17,  19,  21,  23,  20,  22,  24,  26,  28,  25,  27,  29,
+      31,  33,  30,  32,  34,  36,  38,  35,  37,  39,  41,  43,  40,  42,
+      44,  46,  48,  45,  47,  49,  51,  53,  50,  52,  54,  56,  58,  55,
+      57,  59,  61,  63,  60,  62,  64,  66,  68,  65,  67,  69,  71,  73,
+      70,  72,  74,  76,  78,  75,  77,  79,  81,  83,  80,  82,  84,  86,
+      88,  85,  87,  89,  91,  93,  90,  92,  94,  96,  98,  95,  97,  99,
+      101, 103, 100, 102, 104, 106, 108, 105, 107, 109, 111, 113, 110, 112,
+      114, 116, 118, 115, 117, 119, 121, 123};
+    nntrainer::Tensor answer(ref_dim, answer_data);
+    int status = t.add_i(m);
+    EXPECT_EQ(status, ML_ERROR_NONE);
+    EXPECT_EQ(t, answer);
+  }
+  {
+    nntrainer::Tensor t = ranged(3, 2, 4, 5);
+    nntrainer::Tensor m = ranged(1, 2, 1, 1);
+    float answer_data[] = {
+      0,   1,   2,   3,   4,   5,   6,   7,   8,   9,   10,  11,  12,  13,
+      14,  15,  16,  17,  18,  19,  21,  22,  23,  24,  25,  26,  27,  28,
+      29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,  40,  40,  41,
+      42,  43,  44,  45,  46,  47,  48,  49,  50,  51,  52,  53,  54,  55,
+      56,  57,  58,  59,  61,  62,  63,  64,  65,  66,  67,  68,  69,  70,
+      71,  72,  73,  74,  75,  76,  77,  78,  79,  80,  80,  81,  82,  83,
+      84,  85,  86,  87,  88,  89,  90,  91,  92,  93,  94,  95,  96,  97,
+      98,  99,  101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112,
+      113, 114, 115, 116, 117, 118, 119, 120};
+    nntrainer::Tensor answer(ref_dim, answer_data);
+    int status = t.add_i(m);
+    EXPECT_EQ(status, ML_ERROR_NONE);
+    EXPECT_EQ(t, answer);
+  }
+  {
+    nntrainer::Tensor t = ranged(3, 2, 4, 5);
+    nntrainer::Tensor m = ranged(3, 1, 1, 1);
+    float answer_data[] = {
+      0,   1,   2,   3,   4,   5,   6,   7,   8,   9,   10,  11,  12,  13,
+      14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,  26,  27,
+      28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,  41,  42,
+      43,  44,  45,  46,  47,  48,  49,  50,  51,  52,  53,  54,  55,  56,
+      57,  58,  59,  60,  61,  62,  63,  64,  65,  66,  67,  68,  69,  70,
+      71,  72,  73,  74,  75,  76,  77,  78,  79,  80,  82,  83,  84,  85,
+      86,  87,  88,  89,  90,  91,  92,  93,  94,  95,  96,  97,  98,  99,
+      100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113,
+      114, 115, 116, 117, 118, 119, 120, 121};
+    nntrainer::Tensor answer(ref_dim, answer_data);
+    int status = t.add_i(m);
+    EXPECT_EQ(status, ML_ERROR_NONE);
+    EXPECT_EQ(t, answer);
+  }
+  {
+    nntrainer::Tensor t = ranged(3, 2, 4, 5);
+    nntrainer::Tensor m = ranged(1, 1, 1, 1);
+    float answer_data[] = {
```

Comment:
It might be clear when m is set by other number like 1 or 2.

Reply:
Changed. Thanks